tripy/nvtripy/frontend/ops/plugin.py

```diff
@@ -75,3 +78,125 @@ def plugin(
+            (2). 1-D Numpy arrays.
+
+    Returns:
+        The output(s) of the plugin either as a single tensor if there is only one output,
+        or a list of tensors otherwise.
+
+    .. code-block:: python
+        :linenos:
+
+        import triton
+        import triton.language as tl
+
+        import tensorrt.plugin as trtp
+
+        @triton.jit
+        def add_kernel(x_ptr, n_elements, y_ptr, BLOCK_SIZE: tl.constexpr):
+            pid = tl.program_id(0)
+            offsets = pid * BLOCK_SIZE + tl.arange(0, BLOCK_SIZE)
+            mask = offsets < n_elements
+            x = tl.load(x_ptr + offsets, mask=mask)
+            tl.store(y_ptr + offsets, x + 1, mask=mask)
+
+        @trtp.register("example::elemwise_add_plugin")
+        def add_plugin_desc(inp0: trtp.TensorDesc, block_size: int) -> trtp.TensorDesc:
+            return inp0.like()
+
+        @trtp.aot_impl("example::elemwise_add_plugin")
+        def add_plugin_aot_impl(
+            inp0: trtp.TensorDesc, block_size: int, outputs: Tuple[trtp.TensorDesc], tactic: int
+        ) -> Tuple[Union[str, bytes], Union[str, bytes], trtp.KernelLaunchParams, trtp.SymExprs]:
+
+            src = triton.compiler.ASTSource(
+                fn=add_kernel,
+                signature="*fp32,i32,*fp32",
+                constants={
+                    "BLOCK_SIZE": block_size,
+                },
+            )
+
+            compiled_kernel = triton.compile(src)
+
+            launch_params = trtp.KernelLaunchParams()
+            N = inp0.shape_expr.numel()
+            launch_params.grid_x = trtp.cdiv(N, block_size)
+            launch_params.block_x = compiled_kernel.metadata.num_warps * 32
+            launch_params.shared_mem = compiled_kernel.metadata.shared
+
+            extra_args = trtp.SymIntExprs(1)
+            extra_args[0] = trtp.SymInt32(N)
+
+            return compiled_kernel.metadata.name, compiled_kernel.asm["ptx"], launch_params, extra_args
+
+        inp = tp.iota((2, 2))
+        out = tp.plugin(
+            "example::elemwise_add_plugin",
+            [inp],
+            block_size = 256,
+        )
+
+        assert cp.allclose(cp.from_dlpack(out), cp.from_dlpack(inp + 1))
+    """
+
+    namespace, name = op.split("::")
+
+    try:
+        trtp_op = getattr(getattr(trtp.op, namespace), name)
+    except AttributeError:
+        raise_error(f"Plugin {op} not found. Expected a plugin of the form '<namespace>::<name>'.")
+
+    attrs = {}
+    for key, value in kwargs.items():
+        if key not in trtp_op.input_attrs:
+            raise_error(f"Unexpected attribute {key} provided. Expected one of {trtp_op.input_attrs.keys()}.")
+
+        attr_annotation = trtp_op.input_attrs[key]
+
+        if isinstance(value, np.ndarray):
+            kwargs[key] = value.tolist()
+        else:
+            if attr_annotation is not type(value):
+                raise_error(f"Unexpected type '{type(value)}' for attribute '{key}'. Expected '{attr_annotation}'.")
```

Comment:
Can use `type_str_from_arg` and `str_from_type_annotation` here to make sure the display of these types is clean.

Reply:
Done